cns/restserver/ipam.go

```diff
@@ -661,8 +675,15 @@ func (service *HTTPRestService) GetExistingIPConfig(podInfo cns.PodInfo) ([]cns.
 
 // Assigns a pod with all IPs desired
 func (service *HTTPRestService) AssignDesiredIPConfigs(podInfo cns.PodInfo, desiredIPAddresses []string) ([]cns.PodIpInfo, error) {
+	// Sets the number of desired IPs equal to the number of NCs so that we can get one IP per NC
 	numDesiredIPAddresses := len(desiredIPAddresses)
+	// Creates a slice of PodIpInfo with the size as number of NCs to hold the result for assigned IP configs
 	podIPInfo := make([]cns.PodIpInfo, numDesiredIPAddresses)
+	// if there are no NCs on the NNC there will be no IPs in the pool so return error
+	if numDesiredIPAddresses == 0 {
+		//nolint:goerr113 // return error
+		return podIPInfo, fmt.Errorf("no NCs found on the NNC so no IPs are in the pool")
```

Comment:
Is the error message appropriate here? Seems oddly coupled that this method knows about NCs or NNCs. It only knows that it can't do anything meaningful when desired IPs are 0

Reply:
Agree @ramiro-gamarra . There's an urge to provide more insight when returning an error, but usually this causes the error to ultimately tell lies when the facts of that insight change over time. It's better to return an error about exactly what it knows and provide additional contextual information through error wrapping.

Reply:
This should actually be based on `len(service.state.ContainerStatus)` which is the number of NCs. I'll fix this and it will be more accurate